cpp/src/parquet/arrow/reader.h

```diff
@@ -249,6 +249,90 @@ class PARQUET_EXPORT FileReader {
   virtual ::arrow::Status ReadRowGroups(const std::vector<int>& row_groups,
                                         std::shared_ptr<::arrow::Table>* out) = 0;
 
+  using AsyncBatchGenerator =
+      std::function<::arrow::Future<std::shared_ptr<::arrow::RecordBatch>>()>;
+
+  /// \brief Read a single row group from the file
+  ///
+  /// \see ReadRowGroupsAsync for operation details
+  ///
+  /// \param i the index of the row group to read
+  /// \param cpu_executor an executor to use to run CPU tasks
+  /// \param allow_sliced_batches if false, an error is raised if a batch has too much
+  ///                             data for the given batch size.  If true, smaller
+  ///                             batches will be returned instead.
+  virtual AsyncBatchGenerator ReadRowGroupAsync(int i,
```

Comment:
Is it necessary to expose `ReadRowGroupAsync` in addition to `ReadRowGroupsAsync`? One is a trivial call to the other...

Reply:
I was attempting to maintain parity with the synchronous methods above.  I only need one of these four methods and so if you'd prefer I'm happy to scope this down.

Reply:
At least the single-row group `ReadRowGroupAsync` is a trivial redirect to the several-row groups variant, so removing those two would be fine.

Reply:
Sounds good.  I've removed the single-row variants.